src/components/timeline/index.tsx

```diff
@@ -0,0 +1,288 @@
+import React, {useCallback, useMemo, useEffect, useState} from 'react';
+import {StyleSheet, ImageRequireSource, ViewStyle, MeasureOnSuccessCallback, LayoutChangeEvent} from 'react-native';
+import {Colors, Spacings} from '../../style';
+import View from '../view';
+import Icon from '../icon';
+import Text from '../text';
+import Dash from './Dash';
+
+
+const LINE_WIDTH = 2;
+const POINT_SIZE = 12;
+const HALO_WIDTH = 4;
+const HALO_TINT = 70;
+const HOLLO_WIDTH = 2;
+const CONTENT_POINT_SIZE = 20;
+const ICON_SIZE = 12;
+const CONTENT_CONTAINER_PADDINGS = Spacings.s2;
+const POINT_MARGINS = Spacings.s1;
+
+export enum StateTypes {
+  CURRENT = 'current', // default
+  NEXT = 'next',
+  ERROR = 'error',
+  SUCCESS = 'success'
+}
+
+export enum LineTypes {
+  FULL = 'full', // default
+  DASHED = 'dashed'
+}
+
+export enum PointTypes {
+  BULLET = 'bullet', // default
+  HOLLOW = 'hollow',
+  HALO = 'halo'
+}
+
+export type LineProps = {
+  state?: StateTypes;
+  type?: LineTypes;
+  color?: string;
+  /** to mark as entry point */
+  entry?: boolean;
+}
+
+export type PointProps = {
+  state?: StateTypes;
+  type?: PointTypes;
+  color?: string;
+  icon?: ImageRequireSource;
+  label?: number;
+  /** to align point to this view's center */
+  alignmentTargetRef?: React.MutableRefObject<undefined>;
+  /** the target view's top parent view */
+  targetContainerRef?: React.MutableRefObject<undefined>;
```

Comment:
Yes, it is meant to align the point to a specific view. I think that passing two refs is easier than passing the offset as no calculations are involved. And the reason there are two refs instead of one is that `measure` doesn't work on Android and I had to use `measureLayout` instead.

Reply:
To the very least I'd would support both APIs
- For users who don't want to maintain a list of refs then can just pass offset -easy to use, less accurate 
- For users who want a specific position we can support the ref - more complex but more accurate 

In general, the reason I'm less fond of the ref solution is because it means the complex code that has potential for bugs are in our code.

Reply:
I reduced it to one ref - pass the ref of the view you want to align the point to (the container ref is now in the Timeline component).

Reply:
To add support for 'topMargin' is not an easy thing. I suggest seeing if it is necessary after the release. I think passing one target is super simple.

Reply:
Ok